ios-coding-guidelines.md

```diff
@@ -2,24 +2,39 @@
 1. **Never omit specifying the value of string-typed enum cases.** Imagine you create a string-typed enum and omit the values for every case since Swift infers the value from the case name. Now when you refactor one enum case from somewhere in your app, the inferred value also changes, which is not desired i.e. in JSON keys. You would introduce hard to find errors.
-1. **Create all classes as `final`.** This way some optimisations are enabled for these classes which speeds up compilation.
+1. **Avoid subclasses and finalize your classes** Because super- and subclass logic becomes hard to maintain, try to split up logic into objects instead. By marking your classes final, some optimizations are enabled which among other things speeds up compilation.
 1. **Set thickness of lines device dependent.** When you set the thickness of lines to `1.0 / UIScreen.main.scale` then the will appear equally on all devices.
-1. **Use UITextView for multiline text.** If you follow this rule you get text selection and all the features of an UITextView for free. Be sure you set isEditable and isScrollable to `false`.
+1. **Use UITextView for multiline text.** If you have a block of text, UITextField is more practical than a multiline label, especially if you want your content to be selectable. Be sure you set isEditable and isScrollable to `false` if desired.
 1. **Localized-strings used in a framework F must be defined there.** If don't folow this rule, **F** depends on the target/module where the string is defined and thus is not modular/independent anymore. If e.g. another target/module uses a class of **F** that contains a localized string which is outside of **F** the translation could not be found.
 1. **Name selectors with the pattern on<action_name>** Having a unified pattern for naming selectors helps to make code understandable faster. The on* pattern is also used in other programming languages like JavaScript.
-1. **No file header.** Remove comment header for newly created files since it will be always outdated (copyright, spelling errors in file names, etc.).
+1. **Name bools with the pattern isState** This is easy to read and matches the swifty naming conventions in the updated iOS frameworks.
+
+## Framework Extensions
+
+1. **Never set `translatesAutoresizingMaskIntoConstraints` manually.** We have the functions `addTo()` and `addMaximizedTo()` which do that automatically.
+1. **UI Alerts** Use UIAlertController extension to display alerts and action sheets.
+1. **Fonts** Never define a font in code, but rather use or extend the presets in the Fonts class.
+1. **Colors** Never define a color in code, but rather use or extend the UIColor extensions.
```

Comment:
extend the UIColor extensions. > extend the Palette struct and the related UIColor extension.